src/service/extension-location.js

```diff
@@ -28,20 +28,44 @@ import {urls} from '../config';
  */
 let ExtensionInfoDef;
 
+/**
+ * Dynamic AMP components
+ */
+const AMP_DYNAMIC_COMPONENTS = ['amp-geo'];
```

Comment:
This hardcoded list (now in the base runtime) is disconnected from the source of truth that makes a component "dynamic."

Are there many others that are expected to fall into this category? What happens when they do?

Reply:
I don't know the future of how many components will become "dynamic components". I'm hesitant to assume it will always be just `amp-geo`. The rationale for identifying these components is discussed in https://github.com/ampproject/amp-toolbox/issues/515 and https://github.com/ampproject/amp-toolbox/pull/518. Or, tldr;

> Not everyone who self-hosts the AMP runtime has the infrastructure to
> support dynamic components like amp-geo, ref: [AMP cache guidelines](https://github.com/ampproject/amphtml/blob/master/spec/amp-cache-guidelines.md#guidelines-adding-a-new-cache-to-the-amp-ecosystem).

I agree with your 'source of truth' comment. This probably isn't the best place for this. Would it make more sense to introduce `src/dynamic-components.js`?

Reply:
This is resolved in commit d76da0c. Let me know how the changes look to you.